money.go

```diff
@@ -27,12 +27,26 @@ func defaultUnmarshalJSON(m *Money, b []byte) error {
 	if err != nil {
 		return err
 	}
-	ref := New(int64(data["amount"].(float64)), data["currency"].(string))
+
+	amount := int64(data["amount"].(float64))
```

Comment:
> Is it going to panic if we'll try to unmarshal {} (empty JSON) because of the unset keys in the data?

I think so, but I believe it would have before these changes. I'll throw in a fix though, I think it's reasonable to assume `{}` would give you back a zero value instance of `Money`.

> Should we not allow currency as an empty field at all?

Hmm good design question. I guess since it's Go and you can always do `money.Money{}` and not specify anything, IMO it's better to gracefully handle what happens in that situation rather than trying to enforce going through the constructor or specifying a currency. Also feels Go-like to me to have some sort of meaningful zero value since that seems to be the std lib convention. But you're the maintainer/author here, so happy to work with you!